libraries/standard/http/include/http_client.h

```diff
@@ -423,19 +460,29 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * parameter pRequestBodyBuf over the transport. The response is received in
  * #HTTPResponse_t.
  *
+ * The application should close the connection with the server if any
+ * HTTP_SECURITY_ALERT_X errors are returned. Any error found in parsing is
+ * considered a malformed response and therefore a security alert.
+ *
  * TODO: Expand documentation.
  *
  * @param[in] pTransport Transport interface, see #HTTPTransportInterface_t for
  * more information.
  * @param[in] pRequestHeaders Request configuration containing the buffer of
  * headers to send.
- * @param[in] pRequestBodyBuf Request entity body.
+ * @param[in] pRequestBodyBuf Request entity body. This can be NULL.
  * @param[in] reqBodyBufLen The length of the request entity in bytes.
  * @param[in] pResponse The response message and some notable response
  * parameters will be returned here on success.
  *
- * @return #HTTP_SUCCESS if successful, an error code otherwise.
- * TODO: Update for exact error codes returned.
+ * @return One of the following:
+ * - #HTTP_SUCCESS
+ * - #HTTP_INVALID_PARAMETER
+ * - #HTTP_NETWORK_ERROR
+ * - #HTTP_PARTIAL_RESPONSE
+ * - #HTTP_NO_RESPONSE
+ * - #HTTP_INSUFFICIENT_MEMORY
+ * TODO: Add more errors for parsing implementation.
  */
 HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
```

Comment:
Looking at the function again in the API, a `xxx_SendAndReceived` naming would be clearer about the behavior of this function

Reply:
The name of this function has been approved in sign off. It is clear that we are receiving the response since it is in the parameters.